Realm/RLMSchema.mm

```diff
@@ -180,8 +180,13 @@ + (void)registerClasses:(Class *)classes count:(NSUInteger)count {
 
 // schema based on runtime objects
 + (instancetype)sharedSchema {
+    static mach_port_t threadID;
+    if (pthread_mach_thread_np(pthread_self()) == threadID) {
+        @throw RLMException(@"Illegal recursive call of +[%@ %@]. Note: Properties of Swift `Object` classes must not be prepopulated with queried results from a Realm.", self, NSStringFromSelector(_cmd));
```

Comment:
Names can changes. Stringly-encoded type information tends to be not updated accordingly.